shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+                Assert.equal(newValue, manager.get(newKey), "Should return cookie value");
+                
+                // Disable cookies and delete the cookie
+                manager.setEnabled(false);
+                let delResult = manager.del(newKey);
+                Assert.equal(true, delResult, "Delete should return true when cached");
+                
+                // Cookie still exists in storage but deletion is cached
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should still be in storage while disabled");
+                
+                // Enable cookies - cached deletion should be applied
+                manager.setEnabled(true);
+                
+                // Check that the deletion was applied (cookie should have expiry in the past)
+                let cookieValue = this._testCookies[newKey];
+                Assert.ok(cookieValue, "Delete operation should have been applied");
+                Assert.ok(cookieValue.indexOf("expires=Thu, 01 Jan 1970 00:00:01 GMT") !== -1, "Cookie should have past expiry date");
+                Assert.equal("", manager.get(newKey), "Should return empty string after deletion is applied");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - ignore blocked cookies",
+            test: () => {
+                let cookieCfg: ICookieMgrConfig = objExtend(true, {}, this._cookieMgrCfg);
+                cookieCfg.blockedCookies = ["blockedCookie"];
+                
+                let manager = createCookieMgr({
+                    cookieCfg: cookieCfg
+                });
+
+                // Start with cookies disabled
+                manager.setEnabled(false);
+
+                let blockedKey = "blockedCookie";
+                let allowedKey = "allowedCookie";
+                let newValue = newId();
+                
+                // Try to set blocked cookie while disabled - should not be cached
+                let blockedResult = manager.set(blockedKey, newValue);
+                Assert.equal(false, blockedResult, "Setting blocked cookie should return false");
+                Assert.equal("", manager.get(blockedKey), "Should not return cached value for blocked cookie");
+                
+                // Set allowed cookie while disabled - should be cached
+                let allowedResult = manager.set(allowedKey, newValue);
+                Assert.equal(true, allowedResult, "Setting allowed cookie should return true");
+                Assert.equal(newValue, manager.get(allowedKey), "Should return cached value for allowed cookie");
+
+                // Enable cookies - only allowed cookie should be flushed
+                manager.setEnabled(true);
+                
+                Assert.equal(undefined, this._testCookies[blockedKey], "Blocked cookie should not be in storage");
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect